banyan/src/store/mem_store.rs

```diff
@@ -60,17 +60,17 @@ impl<L: Eq + Hash + Copy> MemStore<L> {
 }
 
 impl<L: Eq + Hash + Copy + Send + Sync + 'static> ReadOnlyStore<L> for MemStore<L> {
-    fn get(&self, link: &L) -> anyhow::Result<Box<[u8]>> {
+    fn get(&self, link: &L) -> Result<Box<[u8]>, Error> {
         if let Some(value) = self.get0(link) {
             Ok(value)
         } else {
-            Err(anyhow!("not there"))
+            Err(Error::NotThere)
```

Comment:
This will be relevant also for external implementations, right? So perhaps `Error::BlockNotFound`.

Reply:
So why return an error instead of an `Option` here? A `Option` expresses clearly what happened...